docs/discussion/Ratify v2 design scope.md

```diff
@@ -0,0 +1,165 @@
+Proposal: A more lightweight and extensible framework for Ratify v2
+=======================
+Author: Binbin Li
+
+## Table of Contents
+- [Proposal: A more lightweight and extensible framework for Ratify v2](#proposal-a-more-lightweight-and-extensible-framework-for-ratify-v2)
+  - [Table of Contents](#table-of-contents)
+  - [Background](#background)
+  - [Goal](#goal)
+  - [Scope](#scope)
+    - [Refactoring](#refactoring)
+      - [Cache Refactoring](#cache-refactoring)
+      - [Policy Provider Refactoring](#policy-provider-refactoring)
+      - [Configuration Refactoring](#configuration-refactoring)
+      - [Reorganize the Codebase](#reorganize-the-codebase)
+    - [Deprecations](#deprecations)
+      - [CertificateStore](#certificatestore)
+      - [Legacy Cosign Verifier](#legacy-cosign-verifier)
+      - [License Checker Verifier](#license-checker-verifier)
+      - [`Name` and `Type` fields in verifierReport/configuration](#name-and-type-fields-in-verifierreportconfiguration)
+    - [New Features](#new-features)
+    - [Documentation Update](#documentation-update)
+    - [Unit Test Improvement](#unit-test-improvement)
+
+## Background
+Ratify has reached its first major release in over a year and has been widely adopted by users at scale. However, there are known limitations in the current design and implementation of Ratify v1 make it challenging to extend functionality. This document outlines these known limitations and proposes solutions for a more lightweight and extensible framework.
+
+## Goal
+
+The goal of the v2 design is to improve the current design and implementation to better meet the requirements of the customers.
+
+1. Deprecate outdated features so that customers can have a clear understanding of the features that are still supported.
+2. Refactor the code to make it more maintainable and extensible for new features.
+3. Improve the performance of the system.
+4. Improve the usability of the system.
+5. Reduce the dependencies of the system.
+
+## Scope
+
+Overall, the scope of the v2 design includes the following aspects: system refactoring, deprecations, new features, document update and unit test improvement. We'll discuss each aspect in detail.
```

Comment:
How about attestation support that is coming up on the horizon? Attestations are tricky in that they are synthetic bundle of signatures, envelope, and a verifiable artifact. This may introduce a new first class component in Ratify. Would this be out of scope for v2.0.0?

Reply:
good call! That should be considered in v2.0.0 but will be low prioritized. I feel the problem is that we don't have a clear design for attestation support yet. Basically the v2 prototype should be easy for us to support it without breaking change.